[WIP] HTTP Signatures

allouis opened the pull request:
🍦

sagzy commented on ghost/ghost/src/core/activitypub/http-signature.service.ts:
```diff
@@ -0,0 +1,156 @@
+            requestHeaders.signature
+        );
+
+        if (requestMethod === 'post') {
```
This block is not being called by the tests because of a case mismatch "post" vs "POST"

allouis committed:
Added HTTPSignature service

ref https://linear.app/tryghost/issue/MOM-72

This module handles signing and validating HTTP signatures, which is a core
part of interfacing with ActivityPub enabled servers.

allouis committed:
Added ability for Actor to sign requests

ref https://linear.app/tryghost/issue/MOM-74

This will allow us to generated signed requests for Activites.
